src/Driver.cpp

```diff
@@ -28,10 +30,15 @@ void Driver::open(std::string const& uri)
     openURI(uri);
 }
 
-void Driver::read()
+bool Driver::read()
 {
-    readPacket(&buffer[0], buffer.size());
-    parseMessage(&buffer[4],buffer.size()-4);
+    int msg_size = readPacket(&buffer[0], buffer.size());
+    if(msg_size == KVH_1750_MSG_SIZE)
```

Comment:
> The contract is in the iodrivers_base API. Adding a comment if that is unclear to you is of course a great idea, but testing in the live codepath for a failure of extractPacket is definitely not a good practice. readPacket will only return a packet as sized by extractPacket by contract.

Yes, I will leave a comment just for me.

> Basically what you do in this PR is fix a bug, and then test in the live code whether the bug has been fixed, leaking that fact to your callers. It might not have been your intent, but you not only did create a dead path, you basically created a path that would be taken only if the PR did not fix the bug it's attempting to fix.

No, this line was not supposed to check for a bug that was fixed in the same commit.
I agree that the check, due to the contract in the parent class, is not necessary here.